draft-ietf-tls-tls13.md

```diff
@@ -1817,6 +1817,17 @@ the 0-RTT data without detection. If the client's ephemeral share
 is compromised and client authentication is used, then the attacker
 can impersonate the client on subsequent connections.
 
+It is up to the implementation and application layers above TLS to mitigate
+the impact of these weaknesses.  For example, to support strong client
+authentication and forward secrecy over 0-RTT connections, it is possible to
+issue a new ticket on the first 1-RTT connection, and on every resumed session
+after that.  Each co-located group of servers could ensure that tickets they
+issue are used only once by maintaining a local cache of ticket decryption
+keys.  An accurate clock on the client could ensure freshness of client
+signatures.  In addition it is a good idea to make all request idempotent,
```

Comment:
A traditional proof of possession provides two things: uniqueness and freshness.  The server knows the signature is unique since the server picked a unique challenge to sign.  We can get the same effect by ensuring the servers will never accept the same signed nonce twice.

As you point out, the more difficult trait we want is freshness.  Traditional challenge-response happens in less than a second, proving that the key was available to the client for signing within the last 1 second.  OTP keys have accurate clocks, which is how they are able to show you a pin to type in when prompted.  They are accurate to a few seconds typically.  If a user kept his client certificate in a security key with a similar clock, the key would create the nonce and timestamp when it creates the signature.  The company issuing the keys can be confident that only an attacker with the access and skills to hack the security key could fake the signed nonce and timestamp.  This is not quite as strong as a real proof of possession.  I call it "compelling evidence".

Anyway, with good secure clocks, we can indeed get the freshness property we need.